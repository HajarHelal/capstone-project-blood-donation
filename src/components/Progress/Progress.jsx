/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { PropTypes } from 'prop-types';
import { useTranslation } from 'react-i18next';

function Prgress({ steps }) {
  const { t } = useTranslation();
  return (
    <div className="flex  items-center w-10/12  mb-10">
      <div className="text-base md:text-lg lg:text-xl font-semibold  mx-4 sm:mx-8">
        {t('donate_first_page.step')}
        <span className="mx-2">{`${steps} / 3`}</span>
      </div>
      <div className="flex-1 h-6 sm:h-8 md:h-10 bg-gray-100">
        <div
          className={`h-6 sm:h-8 md:h-10  bg-primary ${steps === 1 ? 'w-4/12' : null} ${
            steps === 2 ? 'w-8/12' : null
          } ${steps === 3 ? 'w-100' : null}`}
        />
      </div>
    </div>
  );
}

Prgress.propTypes = {
  steps: PropTypes.number.isRequired,
};
export default Prgress;
