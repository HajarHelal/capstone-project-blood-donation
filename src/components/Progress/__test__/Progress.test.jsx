import React from 'react';
import renderer from 'react-test-renderer';
import Progress from '../Progress';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer.create(<Progress steps={1} />).toJSON();
  expect(tree).toMatchSnapshot();
});
