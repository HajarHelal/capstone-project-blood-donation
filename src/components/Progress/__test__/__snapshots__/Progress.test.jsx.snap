// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex  items-center w-10/12  mb-10"
>
  <div
    className="text-base md:text-lg lg:text-xl font-semibold  mx-4 sm:mx-8"
  >
    donate_first_page.step
    <span
      className="mx-2"
    >
      1 / 3
    </span>
  </div>
  <div
    className="flex-1 h-6 sm:h-8 md:h-10 bg-gray-100"
  >
    <div
      className="h-6 sm:h-8 md:h-10  bg-primary w-4/12 null null"
    />
  </div>
</div>
`;
